integrations/openzeppelin/contracts.md

```diff
@@ -0,0 +1,235 @@
+ 4. **Access Control** — list all of the available [access control mechanism](https://docs.openzeppelin.com/contracts/4.x/access-control) available for each token standard
+ 5. **Interactive code display** — shows the smart contract code with the configuration as set by the user
+
+![OpenZeppelin Contracts Wizard](/images/openzeppelin/ozwizard-images1.png)
+
+Once you have set up your contract with all the settings and features, it is just as easy as copying and pasting the code into your contract file.
+
+## Deploying OpenZeppelin Contracts on Moonbeam
+
+This section goes through the steps for deploying OpenZeppelin contracts on Moonbeam Currently. It covers the following contracts:
+
+ - ERC20 (fungible tokens)
+ - ERC721 (non-fungible tokens)
+ - ERC1155 (multi-token standard)
+
+All the code of the contracts was obtained using OpenZeppelin [Contract Wizard](https://docs.openzeppelin.com/contracts/4.x/wizard).
+ 
+### Checking Prerequisites
+
+The steps described in this section assume you have [MetaMask](https://metamask.io/) installed and connected to the Moonbase Alpha TestNet. Contract deployment is done using the [Remix IDE](https://remix.ethereum.org/) via the "Injected Web3" environment. You can find corresponding tutorials in the following links:
+
+ - [Interacting with Moonbeam using MetaMask](/integrations/wallets/metamask/)
+ - [Interacting with Moonbeam using Remix](/integrations/remix/)
+
+### Deploying an ERC20 Token
+
+For this example, an ERC20 token will be deployed to Moonbase Alpha. The final code used combines different contracts from OpenZeppelin:
+
+ - **ERC20.sol** — ERC20 token implementation with the optional features from the base interface. Includes the supply mechanism with a `mint` function but needs to be explicitly called from within the main contract
+ - **Ownable.sol** — extension to restrict access to certain functions
+
+The mintable ERC20 OpenZeppelin token contract provides a `mint` function that the owner of the contract can only call. By default, the owner is the contract's deployer address. There is also a premint of `1000` tokens sent to the contract's deployer configured in the `constructor` function.
+
+The first step is to go to [Remix](https://remix.ethereum.org/) and take the following steps:
+
+ 1. Click on the "Create New File" icon and set a file name. For this example, it was set to `ERC20.sol`
+ 2. Make sure the file was created successfully. Click on the file to open it up in the text editor
+ 3. Write your smart contract using the file editor. For this example, the following code was used:
+
+```sol
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract MyToken is ERC20, Ownable {
+    constructor() ERC20("MyToken", "MTK") {
+        _mint(msg.sender, 1000 * 10 ** decimals());
+    }
+
+    function mint(address to, uint256 amount) public onlyOwner {
+        _mint(to, amount);
+    }
+}
+```
```

Comment:
what did you have to do in contract wizard to get this exact contract?

Reply:
Good point, I've added the following text after the code (and before the image):

```
This ERC20 token smart contract was extracted from the [Contract Wizard](#openzeppelin-contract-wizard), setting a premint of `1000` tokens and activating the `Mintable` option.
```